src/monitor/nec_monitor_main.cpp

```diff
@@ -19,31 +19,30 @@
  * along with lo2s.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <filesystem>
+
+#include <lo2s/log.hpp>
 #include <lo2s/monitor/nec_monitor_main.hpp>
 
+extern "C"
+{
+    extern int ve_sysfs_path_info(int nodeid, char* ve_sysfs_path);
+}
+
 namespace lo2s
 {
 namespace nec
 {
 
-std::optional<NecDevice> NecMonitorMain::get_device_of(Thread thread)
-{
-    // /sys/class/ve/ve0 is not device 0, because that would make too much sense
-    // So look the device id up here
-
-    for (int i = 0; i < nodeinfo_.total_node_count; i++)
-    {
-        if (!ve_check_pid(nodeinfo_.nodeid[i], thread.as_pid_t()))
-        {
-            return NecDevice(nodeinfo_.nodeid[i]);
-        }
-    }
-    return std::optional<NecDevice>();
-}
-
 std::vector<Thread> NecMonitorMain::get_tasks_of(NecDevice device)
 {
-    std::ifstream task_stream(fmt::format("/sys/class/ve/ve{}/task_id_all", device.as_int()));
+    char sysfs_path_str[PATH_MAX];
+    memset(sysfs_path_str, 0, PATH_MAX);
+
+    ve_sysfs_path_info(device.as_int(), sysfs_path_str);
+    std::filesystem::path sysfs_path(sysfs_path_str);
+    sysfs_path /= "task_id_all";
```

Comment:
Technically speaking, if sysfs_path_str is allowed to be PATH_MAX, appending `/task_id_all` here produces a path potentially longer when PATH_MAX.

Reply:
I think that is something we have to live with. Hopefully nobody is putting enough NEC VE's into a single computer to outrun PATH_MAX.